cclib/parser/turbomoleparser.py

```diff
@@ -208,6 +209,75 @@ def extract(self, inputfile, line):
+        # scf convergence criterion : increment of total energy < .1000000D-05
+        #                  and increment of one-electron energy < .1000000D-02
+        #
+        # ...
+        # ...
+        #                                              current damping :  0.700
+        # ITERATION  ENERGY          1e-ENERGY        2e-ENERGY     NORM[dD(SAO)]  TOL
+        #   1  -382.34543727790    -1396.8009423     570.56292464    0.000D+00 0.556D-09
+        #                            Exc =   -57.835278090846     N = 69.997494722    
+        #          max. resid. norm for Fia-block=  2.782D-05 for orbital     33a         
+        # ...
+        # ...
+        #                                              current damping :  0.750
+        # ITERATION  ENERGY          1e-ENERGY        2e-ENERGY     NORM[dD(SAO)]  TOL
+        #   3  -382.34575357399    -1396.8009739     570.56263988    0.117D-03 0.319D-09
+        #                            Exc =   -57.835593208072     N = 69.999813370    
+        #          max. resid. norm for Fia-block=  7.932D-06 for orbital     33a         
+        #          max. resid. fock norm         =  8.105D-06 for orbital     33a         
+        #
+        # convergence criteria satisfied after  3 iterations
+        #
+        #
+        #                  ------------------------------------------ 
+        #                 |  total energy      =   -382.34575357399  |
+        #                  ------------------------------------------ 
+        #                 :  kinetic energy    =    375.67398458525  :
+        #                 :  potential energy  =   -758.01973815924  :
+        #                 :  virial theorem    =      1.98255043001  :
+        #                 :  wavefunction norm =      1.00000000000  :
+        #                  .......................................... 
+        if 'scf convergence criterion' in line:
+            total_energy_threshold = self.float(line.split()[-1])
+            one_electron_energy_threshold = self.float(next(inputfile).split()[-1])
+            scftargets = [total_energy_threshold, one_electron_energy_threshold]
+            if not hasattr(self, 'scftargets'):
+                self.scftargets = []
+            self.scftargets.append(scftargets)
+
+            if self.scf_converged:
+                iter_energy = []
+                iter_one_elec_energy = []
+                while 'convergence criteria satisfied' not in line:
+                    if 'ITERATION  ENERGY' in line:
+                        line = next(inputfile)
+                        info = line.split()
+                        iter_energy.append(utils.convertor(self.float(info[1]), 'hartree', 'eV'))
+                        iter_one_elec_energy.append(utils.convertor(self.float(info[2]), 'hartree', 'eV'))
+                    line = next(inputfile)
+
+                scfvalues = []
+                if len(iter_energy) == len(iter_one_elec_energy):

```

Comment:
Isn't this always true based on your code above? Suggest using `assert` to verify that condition holds instead of an `if`-statement.